src/region.js

```diff
@@ -57,25 +57,34 @@ const Region = MarionetteObject.extend({
 
     this.triggerMethod('before:show', this, view, options);
 
-    monitorViewEvents(view);
-
-    this.empty(options);
-
-    // We need to listen for if a view is destroyed in a way other than through the region.
-    // If this happens we need to remove the reference to the currentView since once a view
-    // has been destroyed we can not reuse it.
-    view.on('destroy', this._empty, this);
+    // Assume an attached view is already in the region for pre-existing DOM
+    if (!view._isAttached) {
```

Comment:
I think pre-existing DOM stuff needs its own documentation: https://github.com/marionettejs/backbone.marionette/issues/3128#issuecomment-258726575  and this is only a small portion of what isn't fully documented.